python/pyarrow/array.pxi

```diff
@@ -951,26 +951,43 @@ cdef class Array(_PandasConvertible):
             return values
         return values.astype(dtype)
 
-    def to_numpy(self):
+    def to_numpy(self, zero_copy_only=True, writable=False):
```

Comment:
I think we should raise `ValueError` if both `zero_copy_only` and `writable` are true. This would prevent user errors or surprises.